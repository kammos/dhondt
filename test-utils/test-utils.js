import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { calculatorReducer } from '../src/calculator';


const customRender = (ui, { initialState, store, ...options } = {}) => {
    const actualStore = store ?? configureStore({
        reducer: { calculator: calculatorReducer },
        preloadedState: initialState,
    });

    const Wrapper = ({ children }) => {
        return <Provider store={actualStore}>
            {children}
        </Provider>
    }

    return render(ui, { wrapper: Wrapper, ...options });
}


export * from '@testing-library/react';
export { customRender as render };