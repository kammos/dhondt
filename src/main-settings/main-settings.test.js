/**
 * @jest-environment jsdom
 */
import { render } from 'test-utils';
import { MainSettings } from '.';


describe('MainSettings component', () => {
    it('should match snapshot', () => {
        const ui = render(<MainSettings />);

        expect(ui.container).toMatchSnapshot();
    });
});