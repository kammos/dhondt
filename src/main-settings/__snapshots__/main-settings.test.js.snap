// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSettings component should match snapshot 1`] = `
<div>
  <div
    aria-labelledby="main-settings-label"
    class="fieldset"
    role="group"
  >
    <div
      class="legend"
      id="main-settings-label"
    >
      Main settings
    </div>
    <div
      class="inputs"
    >
      <label
        class="label number"
      >
        <span
          class="labelText"
        >
          Seats
          :
        </span>
        <input
          class="input"
          name="seats"
          type="number"
          value="100"
        />
      </label>
      <label
        class="label number"
      >
        <span
          class="labelText"
        >
          Treshold
          :
        </span>
        <input
          class="input"
          name="treshold"
          type="number"
          value="0"
        />
      </label>
    </div>
  </div>
</div>
`;
